x-pack/plugins/code/public/components/editor/editor.tsx

```diff
@@ -124,8 +123,11 @@ export class EditorComponent extends React.Component<IProps> {
   }
 
   public loadBlame(blames: GitBlame[]) {
+    if (this.blameWidgets) {
```

Comment:
Do you think it's worth to check if the currently rendered blame matches the current file so that we don't need to load the blames again?

Reply:
`loadBlame` only invoked when blames change or toggle show blame

Reply:
as long as toggling the blame in the same file won't trigger additional blame data fetching, it should be fine.